src/box/lua/index.c

```diff
@@ -47,8 +47,10 @@ static uint32_t CTID_STRUCT_ITERATOR_PTR = 0;
 static int
 lbox_insert(lua_State *L)
 {
-	if (lua_gettop(L) != 2 || !lua_isnumber(L, 1))
-		return luaL_error(L, "Usage space:insert(tuple)");
+	if (lua_gettop(L) != 2 || !lua_isnumber(L, 1)) {
+		diag_set(ClientError, ER_PROC_LUA, "Usage space:insert(tuple)");
```

Comment:
Fixed. Also changed `box.error.PROC_LUA` errors to `box.error.ILLEGAL_PARAMS` both for existing `box.error` and where this patch changes `error` to `box.error` in `schema.lua`.

<details>
<summary>diff</summary>

```diff
diff --git a/src/box/lua/index.c b/src/box/lua/index.c
index b5f00220c..32c85e691 100644
--- a/src/box/lua/index.c
+++ b/src/box/lua/index.c
@@ -48,7 +48,7 @@ static int
 lbox_insert(lua_State *L)
 {
 	if (lua_gettop(L) != 2 || !lua_isnumber(L, 1)) {
-		diag_set(ClientError, ER_PROC_LUA, "Usage space:insert(tuple)");
+		diag_set(ClientError, ER_ILLEGAL_PARAMS, "Usage space:insert(tuple)");
 		return luaT_error(L);
 	}
 
@@ -69,7 +69,7 @@ static int
 lbox_replace(lua_State *L)
 {
 	if (lua_gettop(L) != 2 || !lua_isnumber(L, 1)) {
-		diag_set(ClientError, ER_PROC_LUA,
+		diag_set(ClientError, ER_ILLEGAL_PARAMS,
 			 "Usage space:replace(tuple)");
 		return luaT_error(L);
 	}
@@ -93,7 +93,7 @@ lbox_index_update(lua_State *L)
 	if (lua_gettop(L) != 4 || !lua_isnumber(L, 1) || !lua_isnumber(L, 2) ||
 	    (lua_type(L, 3) != LUA_TTABLE && luaT_istuple(L, 3) == NULL) ||
 	    (lua_type(L, 4) != LUA_TTABLE && luaT_istuple(L, 4) == NULL)) {
-		diag_set(ClientError, ER_PROC_LUA,
+		diag_set(ClientError, ER_ILLEGAL_PARAMS,
 			 "Usage index:update(key, ops)");
 		return luaT_error(L);
 	}
@@ -125,7 +125,7 @@ lbox_upsert(lua_State *L)
 	if (lua_gettop(L) != 3 || !lua_isnumber(L, 1) ||
 	    (lua_type(L, 2) != LUA_TTABLE && luaT_istuple(L, 2) == NULL) ||
 	    (lua_type(L, 3) != LUA_TTABLE && luaT_istuple(L, 3) == NULL)) {
-		diag_set(ClientError, ER_PROC_LUA,
+		diag_set(ClientError, ER_ILLEGAL_PARAMS,
 			 "Usage space:upsert(tuple_key, ops)");
 		return luaT_error(L);
 	}
@@ -155,7 +155,8 @@ lbox_index_delete(lua_State *L)
 {
 	if (lua_gettop(L) != 3 || !lua_isnumber(L, 1) || !lua_isnumber(L, 2) ||
 	    (lua_type(L, 3) != LUA_TTABLE && luaT_istuple(L, 3) == NULL)) {
-		diag_set(ClientError, ER_PROC_LUA, "Usage space:delete(key)");
+		diag_set(ClientError, ER_ILLEGAL_PARAMS,
+			 "Usage space:delete(key)");
 		return luaT_error(L);
 	}
 
@@ -178,7 +179,7 @@ lbox_index_random(lua_State *L)
 {
 	if (lua_gettop(L) != 3 || !lua_isnumber(L, 1) || !lua_isnumber(L, 2) ||
 	    !lua_isnumber(L, 3)) {
-		diag_set(ClientError, ER_PROC_LUA,
+		diag_set(ClientError, ER_ILLEGAL_PARAMS,
 			 "Usage index.random(space_id, index_id, rnd)");
 		return luaT_error(L);
 	}
@@ -197,7 +198,7 @@ static int
 lbox_index_get(lua_State *L)
 {
 	if (lua_gettop(L) != 3 || !lua_isnumber(L, 1) || !lua_isnumber(L, 2)) {
-		diag_set(ClientError, ER_PROC_LUA,
+		diag_set(ClientError, ER_ILLEGAL_PARAMS,
 			 "Usage index.get(space_id, index_id, key)");
 		return luaT_error(L);
 	}
@@ -220,7 +221,7 @@ static int
 lbox_index_min(lua_State *L)
 {
 	if (lua_gettop(L) != 3 || !lua_isnumber(L, 1) || !lua_isnumber(L, 2)) {
-		diag_set(ClientError, ER_PROC_LUA,
+		diag_set(ClientError, ER_ILLEGAL_PARAMS,
 			 "usage index.min(space_id, index_id, key)");
 		return luaT_error(L);
 	}
@@ -243,7 +244,7 @@ static int
 lbox_index_max(lua_State *L)
 {
 	if (lua_gettop(L) != 3 || !lua_isnumber(L, 1) || !lua_isnumber(L, 2)) {
-		diag_set(ClientError, ER_PROC_LUA,
+		diag_set(ClientError, ER_ILLEGAL_PARAMS,
 			 "usage index.max(space_id, index_id, key)");
 		return luaT_error(L);
 	}
@@ -267,7 +268,7 @@ lbox_index_count(lua_State *L)
 {
 	if (lua_gettop(L) != 4 || !lua_isnumber(L, 1) || !lua_isnumber(L, 2) ||
 	    !lua_isnumber(L, 3)) {
-		diag_set(ClientError, ER_PROC_LUA,
+		diag_set(ClientError, ER_ILLEGAL_PARAMS,
 			 "usage index.count(space_id, index_id, "
 			 "iterator, key)");
 		return luaT_error(L);
@@ -309,7 +310,7 @@ lbox_index_iterator(lua_State *L)
 {
 	if (lua_gettop(L) != 5 || !lua_isnumber(L, 1) || !lua_isnumber(L, 2) ||
 	    !lua_isnumber(L, 3)) {
-		diag_set(ClientError, ER_PROC_LUA,
+		diag_set(ClientError, ER_ILLEGAL_PARAMS,
 			 "usage index.iterator(space_id, index_id, "
 			 "type, key, after)");
 		return luaT_error(L);
@@ -349,7 +350,7 @@ lbox_iterator_next(lua_State *L)
 {
 	/* first argument is key buffer */
 	if (lua_gettop(L) < 1 || lua_type(L, 1) != LUA_TCDATA) {
-		diag_set(ClientError, ER_PROC_LUA, "Usage: next(param, state)");
+		diag_set(ClientError, ER_ILLEGAL_PARAMS, "Usage: next(param, state)");
 		return luaT_error(L);
 	}
 
@@ -357,7 +358,7 @@ lbox_iterator_next(lua_State *L)
 	uint32_t ctypeid;
 	void *data = luaL_checkcdata(L, 1, &ctypeid);
 	if (ctypeid != CTID_STRUCT_ITERATOR_PTR) {
-		diag_set(ClientError, ER_PROC_LUA, "Usage: next(param, state)");
+		diag_set(ClientError, ER_ILLEGAL_PARAMS, "Usage: next(param, state)");
 		return luaT_error(L);
 	}
 
@@ -386,7 +387,7 @@ static int
 lbox_index_stat(lua_State *L)
 {
 	if (lua_gettop(L) != 2 || !lua_isnumber(L, 1) || !lua_isnumber(L, 2)) {
-		diag_set(ClientError, ER_PROC_LUA,
+		diag_set(ClientError, ER_ILLEGAL_PARAMS,
 			 "usage index.info(space_id, index_id)");
 		return luaT_error(L);
 	}
@@ -405,7 +406,7 @@ static int
 lbox_index_compact(lua_State *L)
 {
 	if (lua_gettop(L) != 2 || !lua_isnumber(L, 1) || !lua_isnumber(L, 2)) {
-		diag_set(ClientError, ER_PROC_LUA,
+		diag_set(ClientError, ER_ILLEGAL_PARAMS,
 			 "usage index.compact(space_id, index_id)");
 		return luaT_error(L);
 	}
diff --git a/src/box/lua/schema.lua b/src/box/lua/schema.lua
index 02ddb26b7..14cc5a365 100644
--- a/src/box/lua/schema.lua
+++ b/src/box/lua/schema.lua
@@ -1743,7 +1743,7 @@ box.schema.index.alter = atomic_wrapper(function(space_id, index_id, options)
             end
         end
         if not can_update then
-            box.error(box.error.PROC_LUA,
+            box.error(box.error.ILLEGAL_PARAMS,
                       "Don't know how to update both id and" ..
                        cant_update_fields)
         end
@@ -1868,7 +1868,7 @@ local iterator_gen = function(param, state) -- luacheck: no unused args
         Please check out http://www.lua.org/pil/7.3.html for details.
     --]]
     if not ffi.istype(iterator_t, state) then
-        box.error(box.error.PROC_LUA, 'Usage: next(param, state)')
+        box.error(box.error.ILLEGAL_PARAMS, 'Usage: next(param, state)')
     end
     -- next() modifies state in-place
     if builtin.box_iterator_next(state, ptuple) ~= 0 then
@@ -1888,7 +1888,7 @@ local port_c = ffi.cast('struct port_c *', port)
 local function check_space_arg(space, method)
     if type(space) ~= 'table' or (space.id == nil and space.name == nil) then
         local fmt = 'Use space:%s(...) instead of space.%s(...)'
-        box.error(box.error.PROC_LUA, string.format(fmt, method, method))
+        box.error(box.error.ILLEGAL_PARAMS, string.format(fmt, method, method))
     end
 end
 box.internal.check_space_arg = check_space_arg -- for net.box
@@ -1907,7 +1907,7 @@ end
 local function check_index_arg(index, method)
     if type(index) ~= 'table' or (index.id == nil and index.name == nil) then
         local fmt = 'Use index:%s(...) instead of index.%s(...)'
-        box.error(box.error.PROC_LUA, string.format(fmt, method, method))
+        box.error(box.error.ILLEGAL_PARAMS, string.format(fmt, method, method))
     end
 end
 box.internal.check_index_arg = check_index_arg -- for net.box
@@ -2601,7 +2601,7 @@ end
 base_index_mt.alter = function(index, options)
     check_index_arg(index, 'alter')
     if index.id == nil or index.space_id == nil then
-        box.error(box.error.PROC_LUA, "Usage: index:alter{opts}")
+        box.error(box.error.ILLEGAL_PARAMS, "Usage: index:alter{opts}")
     end
     return box.schema.index.alter(index.space_id, index.id, options)
 end
@@ -2699,7 +2699,7 @@ space_mt.upsert = function(space, tuple_key, ops, deprecated)
         local msg = "Error: extra argument in upsert call: "
         msg = msg .. tostring(deprecated)
         msg = msg .. ". Usage :upsert(tuple, operations)"
-        box.error(box.error.PROC_LUA, msg)
+        box.error(box.error.ILLEGAL_PARAMS, msg)
     end
     return internal.upsert(space.id, tuple_key, ops);
 end
@@ -3295,7 +3295,7 @@ end
 local function check_func_arg(func, method)
     if type(func) ~= 'table' or func.name == nil then
         local fmt = 'Use func:%s(...) instead of func.%s(...)'
-        box.error(box.error.PROC_LUA, string.format(fmt, method, method))
+        box.error(box.error.ILLEGAL_PARAMS, string.format(fmt, method, method))
     end
 end
 
@@ -3310,7 +3310,7 @@ func_mt.call = function(func, args)
     check_func_arg(func, 'call')
     args = args or {}
     if type(args) ~= 'table' then
-        box.error(box.error.PROC_LUA, 'Use func:call(table)')
+        box.error(box.error.ILLEGAL_PARAMS, 'Use func:call(table)')
     end
     return box.schema.func.call(func.name, unpack(args, 1, table.maxn(args)))
 end
@@ -3425,7 +3425,8 @@ end
 
 box.schema.user.passwd = function(name, new_password)
     if name == nil then
-        box.error(box.error.PROC_LUA, "Usage: box.schema.user.passwd([user,] password)")
+        box.error(box.error.ILLEGAL_PARAMS,
+                  "Usage: box.schema.user.passwd([user,] password)")
     end
     if new_password == nil then
         -- change password for current user
@@ -3919,7 +3920,8 @@ setmetatable(box.space, { __serialize = box_space_mt })
 local function check_read_view_arg(rv, method)
     if type(rv) ~= 'table' then
         local fmt = 'Use read_view:%s(...) instead of read_view.%s(...)'
-        box.error(box.error.PROC_LUA, string.format(fmt, method, method), 3)
+        box.error(box.error.ILLEGAL_PARAMS,
+                  string.format(fmt, method, method), 3)
     end
 end
 
diff --git a/test/box-luatest/box_read_ffi_disable_test.lua b/test/box-luatest/box_read_ffi_disable_test.lua
index 8b5eee54c..66d77869b 100644
--- a/test/box-luatest/box_read_ffi_disable_test.lua
+++ b/test/box-luatest/box_read_ffi_disable_test.lua
@@ -29,8 +29,8 @@ end)
 g.test_min = function(cg)
     cg.server:exec(function()
         local s = box.space.test
-        t.assert_error_msg_content_equals(
-            "Use index:min(...) instead of index.min(...)",
+        t.assert_error_msg_equals(
+            "Illegal parameters, Use index:min(...) instead of index.min(...)",
             s.index.primary.min)
         t.assert_equals(s.index.primary:min(), {1, 1})
         t.assert_equals(s.index.primary:min(2), {2, 5})
@@ -42,8 +42,8 @@ end
 g.test_max = function(cg)
     cg.server:exec(function()
         local s = box.space.test
-        t.assert_error_msg_content_equals(
-            "Use index:max(...) instead of index.max(...)",
+        t.assert_error_msg_equals(
+            "Illegal parameters, Use index:max(...) instead of index.max(...)",
             s.index.primary.max)
         t.assert_equals(s.index.primary:max(), {10, 1})
         t.assert_equals(s.index.primary:max(2), {2, 5})
@@ -55,9 +55,9 @@ end
 g.test_random = function(cg)
     cg.server:exec(function()
         local s = box.space.test
-        t.assert_error_msg_content_equals(
-            "Use index:random(...) instead of index.random(...)",
-            s.index.primary.random)
+        t.assert_error_msg_equals(
+            "Illegal parameters, Use index:random(...) instead of " ..
+            "index.random(...)", s.index.primary.random)
         t.assert_equals(s.index.primary:random(1), {2, 5})
         t.assert_equals(s.index.primary:random(2), {3, 2})
         t.assert_equals(s.index.secondary:random(3), {8, 2})
@@ -68,13 +68,13 @@ end
 g.test_get = function(cg)
     cg.server:exec(function()
         local s = box.space.test
-        t.assert_error_msg_content_equals(
-            "Use index:get(...) instead of index.get(...)",
+        t.assert_error_msg_equals(
+            "Illegal parameters, Use index:get(...) instead of index.get(...)",
             s.index.primary.get)
-        t.assert_error_msg_content_equals(
+        t.assert_error_msg_equals(
             "Invalid key part count in an exact match (expected 1, got 0)",
             s.index.primary.get, s.index.primary)
-        t.assert_error_msg_content_equals(
+        t.assert_error_msg_equals(
             "Get() doesn't support partial keys and non-unique indexes",
             s.index.secondary.get, s.index.secondary, 1)
         t.assert_equals(s:get(1), {1, 1})
@@ -85,10 +85,10 @@ end
 g.test_select = function(cg)
     cg.server:exec(function()
         local s = box.space.test
-        t.assert_error_msg_content_equals(
-            "Use index:select(...) instead of index.select(...)",
-            s.index.primary.select)
-        t.assert_error_msg_content_equals(
+        t.assert_error_msg_equals(
+            "Illegal parameters, Use index:select(...) instead of " ..
+            "index.select(...)", s.index.primary.select)
+        t.assert_error_msg_equals(
             "Unknown iterator type 'foo'",
             s.index.primary.select, s.index.primary,  {}, {iterator = 'foo'})
         t.assert_equals(s:select(2, {iterator = 'lt'}), {{1, 1}})
@@ -101,14 +101,15 @@ g.test_pairs = function(cg)
     cg.server:exec(function()
         local fun = require('fun')
         local s = box.space.test
-        t.assert_error_msg_content_equals(
-            "Use index:pairs(...) instead of index.pairs(...)",
-            s.index.primary.pairs)
-        t.assert_error_msg_content_equals(
+        t.assert_error_msg_equals(
+            "Illegal parameters, Use index:pairs(...) instead of " ..
+            "index.pairs(...)", s.index.primary.pairs)
+        t.assert_error_msg_equals(
             "Unknown iterator type 'foo'",
             s.index.primary.pairs, s.index.primary,  {}, {iterator = 'foo'})
         local gen = s.index.secondary:pairs(2)
-        t.assert_error_msg_equals("Usage: next(param, state)", gen)
+        t.assert_error_msg_equals(
+            "Illegal parameters, Usage: next(param, state)", gen)
         t.assert_equals(fun.totable(s:pairs(2, {iterator = 'lt'})), {{1, 1}})
         t.assert_equals(fun.totable(s.index.primary:pairs(1)), {{1, 1}})
         t.assert_equals(fun.totable(s.index.secondary:pairs(1)),
diff --git a/test/box-luatest/gh_8260_system_read_view_list_test.lua b/test/box-luatest/gh_8260_system_read_view_list_test.lua
index 0314f7f72..6fdb3cb58 100644
--- a/test/box-luatest/gh_8260_system_read_view_list_test.lua
+++ b/test/box-luatest/gh_8260_system_read_view_list_test.lua
@@ -29,11 +29,11 @@ g_object.test_usage = function(cg)
     cg.server:exec(function()
         local rv = rawget(_G, 'test_rv')
         t.assert_error_msg_equals(
-            'Use read_view:info(...) instead of read_view.info(...)',
-            rv.info)
+            'Illegal parameters, Use read_view:info(...) instead of ' ..
+            'read_view.info(...)', rv.info)
         t.assert_error_msg_equals(
-            'Use read_view:close(...) instead of read_view.close(...)',
-            rv.close)
+            'Illegal parameters, Use read_view:close(...) instead of ' ..
+            'read_view.close(...)', rv.close)
     end)
 end
 
diff --git a/test/box/access.result b/test/box/access.result
index dbd40f2c4..7a908c016 100644
--- a/test/box/access.result
+++ b/test/box/access.result
@@ -554,7 +554,7 @@ box.schema.user.passwd('invalid_user', 'some_password')
 ...
 box.schema.user.passwd()
 ---
-- error: 'Usage: box.schema.user.passwd([user,] password)'
+- error: 'Illegal parameters, Usage: box.schema.user.passwd([user,] password)'
 ...
 session.su('user1')
 ---
diff --git a/test/box/alter_limits.result b/test/box/alter_limits.result
index 96897e687..79814c750 100644
--- a/test/box/alter_limits.result
+++ b/test/box/alter_limits.result
@@ -689,7 +689,7 @@ index = s:create_index('primary', { type = 'hash' })
 -- correct error on misuse of alter
 s.index.primary.alter({unique=false})
 ---
-- error: Use index:alter(...) instead of index.alter(...)
+- error: Illegal parameters, Use index:alter(...) instead of index.alter(...)
 ...
 s.index.primary:alter({unique=false})
 ---
diff --git a/test/box/function1.result b/test/box/function1.result
index 277dee7d5..cce143d49 100644
--- a/test/box/function1.result
+++ b/test/box/function1.result
@@ -545,11 +545,11 @@ func = box.func.divide
 ...
 func.call({4, 2})
 ---
-- error: Use func:call(...) instead of func.call(...)
+- error: Illegal parameters, Use func:call(...) instead of func.call(...)
 ...
 func:call(4, 2)
 ---
-- error: Use func:call(table)
+- error: Illegal parameters, Use func:call(table)
 ...
 func:call()
 ---
@@ -595,7 +595,7 @@ func
 ...
 func.drop()
 ---
-- error: Use func:drop(...) instead of func.drop(...)
+- error: Illegal parameters, Use func:drop(...) instead of func.drop(...)
 ...
 box.func.divide
 ---
@@ -621,7 +621,7 @@ func = box.func["function1.divide"]
 ...
 func:call(4, 2)
 ---
-- error: Use func:call(table)
+- error: Illegal parameters, Use func:call(table)
 ...
 func:call()
 ---
diff --git a/test/box/lua.result b/test/box/lua.result
index 05eda886a..5710d74f1 100644
--- a/test/box/lua.result
+++ b/test/box/lua.result
@@ -854,143 +854,143 @@ pk = space:create_index('primary')
 ...
 space.len()
 ---
-- error: Use space:len(...) instead of space.len(...)
+- error: Illegal parameters, Use space:len(...) instead of space.len(...)
 ...
 space.count({}, {iterator = 'EQ'})
 ---
-- error: Use space:count(...) instead of space.count(...)
+- error: Illegal parameters, Use space:count(...) instead of space.count(...)
 ...
 space.bsize()
 ---
-- error: Use space:bsize(...) instead of space.bsize(...)
+- error: Illegal parameters, Use space:bsize(...) instead of space.bsize(...)
 ...
 space.get({1})
 ---
-- error: Use space:get(...) instead of space.get(...)
+- error: Illegal parameters, Use space:get(...) instead of space.get(...)
 ...
 space.select({}, {iterator = 'GE'})
 ---
-- error: Use space:select(...) instead of space.select(...)
+- error: Illegal parameters, Use space:select(...) instead of space.select(...)
 ...
 space.insert({1, 2, 3})
 ---
-- error: Use space:insert(...) instead of space.insert(...)
+- error: Illegal parameters, Use space:insert(...) instead of space.insert(...)
 ...
 space.replace({1, 2, 3})
 ---
-- error: Use space:replace(...) instead of space.replace(...)
+- error: Illegal parameters, Use space:replace(...) instead of space.replace(...)
 ...
 space.put({1, 2, 3})
 ---
-- error: Use space:replace(...) instead of space.replace(...)
+- error: Illegal parameters, Use space:replace(...) instead of space.replace(...)
 ...
 space.update({1}, {})
 ---
-- error: Use space:update(...) instead of space.update(...)
+- error: Illegal parameters, Use space:update(...) instead of space.update(...)
 ...
 space.upsert({1, 2, 3}, {})
 ---
-- error: Use space:upsert(...) instead of space.upsert(...)
+- error: Illegal parameters, Use space:upsert(...) instead of space.upsert(...)
 ...
 space.delete({1})
 ---
-- error: Use space:delete(...) instead of space.delete(...)
+- error: Illegal parameters, Use space:delete(...) instead of space.delete(...)
 ...
 space.auto_increment({'hello'})
 ---
-- error: Use space:auto_increment(...) instead of space.auto_increment(...)
+- error: Illegal parameters, Use space:auto_increment(...) instead of space.auto_increment(...)
 ...
 space.pairs({}, {iterator = 'EQ'})
 ---
-- error: Use space:pairs(...) instead of space.pairs(...)
+- error: Illegal parameters, Use space:pairs(...) instead of space.pairs(...)
 ...
 space.truncate()
 ---
-- error: Use space:truncate(...) instead of space.truncate(...)
+- error: Illegal parameters, Use space:truncate(...) instead of space.truncate(...)
 ...
 space.format({})
 ---
-- error: Use space:format(...) instead of space.format(...)
+- error: Illegal parameters, Use space:format(...) instead of space.format(...)
 ...
 space.drop()
 ---
-- error: Use space:drop(...) instead of space.drop(...)
+- error: Illegal parameters, Use space:drop(...) instead of space.drop(...)
 ...
 space.rename()
 ---
-- error: Use space:rename(...) instead of space.rename(...)
+- error: Illegal parameters, Use space:rename(...) instead of space.rename(...)
 ...
 space.create_index('secondary')
 ---
-- error: Use space:create_index(...) instead of space.create_index(...)
+- error: Illegal parameters, Use space:create_index(...) instead of space.create_index(...)
 ...
 space.run_triggers(false)
 ---
-- error: Use space:run_triggers(...) instead of space.run_triggers(...)
+- error: Illegal parameters, Use space:run_triggers(...) instead of space.run_triggers(...)
 ...
 pk.len()
 ---
-- error: Use index:len(...) instead of index.len(...)
+- error: Illegal parameters, Use index:len(...) instead of index.len(...)
 ...
 pk.bsize()
 ---
-- error: Use index:bsize(...) instead of index.bsize(...)
+- error: Illegal parameters, Use index:bsize(...) instead of index.bsize(...)
 ...
 pk.min()
 ---
-- error: Use index:min(...) instead of index.min(...)
+- error: Illegal parameters, Use index:min(...) instead of index.min(...)
 ...
 pk.min({})
 ---
-- error: Use index:min(...) instead of index.min(...)
+- error: Illegal parameters, Use index:min(...) instead of index.min(...)
 ...
 pk.max()
 ---
-- error: Use index:max(...) instead of index.max(...)
+- error: Illegal parameters, Use index:max(...) instead of index.max(...)
 ...
 pk.max({})
 ---
-- error: Use index:max(...) instead of index.max(...)
+- error: Illegal parameters, Use index:max(...) instead of index.max(...)
 ...
 pk.random(42)
 ---
-- error: Use index:random(...) instead of index.random(...)
+- error: Illegal parameters, Use index:random(...) instead of index.random(...)
 ...
 pk.pairs({}, {iterator = 'EQ'})
 ---
-- error: Use index:pairs(...) instead of index.pairs(...)
+- error: Illegal parameters, Use index:pairs(...) instead of index.pairs(...)
 ...
 pk.count({}, {iterator = 'EQ'})
 ---
-- error: Use index:count(...) instead of index.count(...)
+- error: Illegal parameters, Use index:count(...) instead of index.count(...)
 ...
 pk.get({1})
 ---
-- error: Use index:get(...) instead of index.get(...)
+- error: Illegal parameters, Use index:get(...) instead of index.get(...)
 ...
 pk.select({}, {iterator = 'GE'})
 ---
-- error: Use index:select(...) instead of index.select(...)
+- error: Illegal parameters, Use index:select(...) instead of index.select(...)
 ...
 pk.update({1}, {})
 ---
-- error: Use index:update(...) instead of index.update(...)
+- error: Illegal parameters, Use index:update(...) instead of index.update(...)
 ...
 pk.delete({1})
 ---
-- error: Use index:delete(...) instead of index.delete(...)
+- error: Illegal parameters, Use index:delete(...) instead of index.delete(...)
 ...
 pk.drop()
 ---
-- error: Use index:drop(...) instead of index.drop(...)
+- error: Illegal parameters, Use index:drop(...) instead of index.drop(...)
 ...
 pk.rename("newname")
 ---
-- error: Use index:rename(...) instead of index.rename(...)
+- error: Illegal parameters, Use index:rename(...) instead of index.rename(...)
 ...
 pk.alter({})
 ---
-- error: Use index:alter(...) instead of index.alter(...)
+- error: Illegal parameters, Use index:alter(...) instead of index.alter(...)
 ...
 space:drop()
 ---
diff --git a/test/box/net.box_incorrect_iterator_gh-841.result b/test/box/net.box_incorrect_iterator_gh-841.result
index b721d4f2b..d85d6f5d0 100644
--- a/test/box/net.box_incorrect_iterator_gh-841.result
+++ b/test/box/net.box_incorrect_iterator_gh-841.result
@@ -114,7 +114,7 @@ cn.space.net_box_test_space:insert{234, 1,2,3}
 ...
 cn.space.net_box_test_space.insert{234, 1,2,3}
 ---
-- error: Use space:insert(...) instead of space.insert(...)
+- error: Illegal parameters, Use space:insert(...) instead of space.insert(...)
 ...
 cn.space.net_box_test_space:replace{354, 1,2,3}
 ---
diff --git a/test/box/update.result b/test/box/update.result
index 5705179ec..2e3989e26 100644
--- a/test/box/update.result
+++ b/test/box/update.result
@@ -464,7 +464,7 @@ s:insert{1, 2, 3}
 ...
 s:update({1})
 ---
-- error: Usage index:update(key, ops)
+- error: Illegal parameters, Usage index:update(key, ops)
 ...
 s:update({1}, {'=', 1, 1})
 ---
@@ -813,7 +813,7 @@ s:update({0}, {{'+', 2, -0x4000000000000001ll}})  -- overflow
 -- some wrong updates --
 s:update({0}, 0)
 ---
-- error: Usage index:update(key, ops)
+- error: Illegal parameters, Usage index:update(key, ops)
 ...
 s:update({0}, {'+', 2, 2})
 ---

```

</details>